tests/manual-tests/lua-cli.js

```diff
@@ -8,54 +8,275 @@ const linit        = require('../../src/linit.js');
+        i = process.argv.length;
+        break;
+    case 'E':
+        has_E = true;
+        break;
+    case 'i':
+        has_i = true;
+        /* (-i implies -v) */
+        /* falls through */
+    case 'v':
+        if (process.argv[i].length > 2) {
+            /* invalid option */
+            print_usage(process.argv[script]);
+            return process.exit(1);
+        }
+        has_v = true;
+        break;
+    case 'e':
+        has_e = true;
+        /* falls through */
+    case 'l':  /* both options need an argument */
+        if (process.argv[i].length < 2) {  /* no concatenated argument? */
+            i++;  /* try next 'process.argv' */
+            if (process.argv.length <= i || process.argv[i][0] === '-') {
+                /* no next argument or it is another option */
+                print_usage(process.argv[script]);
+                return process.exit(1);
+            }
+        }
+        break;
+    default:  /* invalid option */
+        print_usage(process.argv[script]);
+        return process.exit(1);
+    }
+}
 
-for (;;) {
-    let input = readlineSync.prompt();
+if (has_v)
+    console.log(lua.FENGARI_COPYRIGHT);
 
-    if (input.length === 0)
-        continue;
+if (has_E) {
+    /* signal for libraries to ignore env. vars. */
+    lapi.lua_pushboolean(L, 1);
+    lapi.lua_setfield(L, lua.LUA_REGISTRYINDEX, lua.to_luastring("LUA_NOENV"));
+}
 
-    let status;
-    {
-        let buffer = lua.to_luastring("return " + input);
-        status = lauxlib.luaL_loadbuffer(L, buffer, buffer.length, stdin);
+/* open standard libraries */
+linit.luaL_openlibs(L);
+
+/* create table 'arg' */
+lapi.lua_createtable(L, process.argv.length - (script + 1), script + 1);
+for (let i = 0; i < process.argv.length; i++) {
+    lapi.lua_pushliteral(L, process.argv[i]);
+    lapi.lua_seti(L, -2, i - script); /* TODO: rawseti */
+}
+lapi.lua_setglobal(L, lua.to_luastring("arg"));
+
+if (!has_E) {
+    /* run LUA_INIT */
+    let name = "LUA_INIT"+lua.LUA_VERSUFFIX;
+    let init = process.env[name];
+    if (!init) {
+        name = "LUA_INIT";
+        init = process.env[name];
     }
-    if (status !== lua.thread_status.LUA_OK) {
-        lapi.lua_pop(L, 1);
-        let buffer = lua.to_luastring(input);
-        if (lauxlib.luaL_loadbuffer(L, buffer, buffer.length, stdin) === lua.thread_status.LUA_OK) {
-            status = lua.thread_status.LUA_OK;
+    if (init) {
+        let status;
+        if (init[0] === '@') {
+            status = dofile(L, init.substring(1));
+        } else {
+            status = dostring(L, init, name);
+        }
+        if (status !== lua.thread_status.LUA_OK) {
+            return process.exit(1);
         }
     }
-    if (status !== lua.thread_status.LUA_OK) {
-        lauxlib.lua_writestringerror(`${lapi.lua_tojsstring(L, -1)}\n`);
-        continue;
+}
+
+/* execute arguments -e and -l */
+for (let i = 1; i < script; i++) {
+    let option = process.argv[i][1];
```

Comment:
Howso? If there is a argument that starts doesn't start with `-` it will be at an index > `script`